### PR TITLE
Move base64 constant-time functions to the new module

### LINKED ISSUE #4926: Move base64 constant-time functions to the new module
https://github.com/ARMmbed/mbedtls/issues/3649 (PR: https://github.com/ARMmbed/mbedtls/pull/4866) moves constant-time functions to a new module. https://github.com/ARMmbed/mbedtls/issues/4814 (PR: https://github.com/ARMmbed/mbedtls/pull/4835) rewrites base64 constant-time functions. To allow the two to proceed in parallel, the base64 work will happen inside `base64.c` and the main constant-time PR not touch the base64 functions.

The goal of this task is to move base64 constant-time functions to the new module, once both https://github.com/ARMmbed/mbedtls/issues/3649 and https://github.com/ARMmbed/mbedtls/issues/4814 are resolved.

### PR DESCRIPTION
**Pull request by gabor-mezei-arm**
## Description
The goal of this task is to move base64 constant-time functions to the new constant-time module.

Resolves #4926 

## Requires Backporting
- [x] 2.x #5284

## Todos
- [x] Documentation
- [x] Backported

### CODE REVIEW

**Review by davidhorstmann-arm — CHANGES_REQUESTED** [review]
One question, LGTM otherwise.

File: library/constant_time_internal.h [hunk 1]
```diff
@@ -167,6 +167,35 @@ void mbedtls_ct_mpi_uint_cond_assign( size_t n,
...
+ * but not EBCDIC).
+ *
+ * \param c     A base64 digit.
+ *
+ * \return      The value of the base64 digit \p c.
+ */
+signed char mbedtls_ct_base64_dec_value( unsigned char c );
+
+#endif /* MBEDTLS_BASE64_C */
+
```
[the remark this answers is not in the document]
Comment (gabor-mezei-arm) [reply library/constant_time_internal.h:198]: They are called from the `base64.c`, so the function declarations must always be available. The `constant_time_invasive.h` contains only the locally used functions which needs to be tested and these are only available when testing.
The `mbedtls_ct_base64_enc_char` and `mbedtls_ct_base64_dec_value` were local functions in `base64.c` and that is why were in the `base64_invasive.h`. After moving to the constant-time module they are referenced from another module so cannot be local functions.

**Review by davidhorstmann-arm — APPROVED** [review]
LGTM - thanks!

**Review by mpg — COMMENTED** [review]
I understood the goal of the task differently, so I'd like this point to be resolved before I can approve this PR. https://github.com/ARMmbed/mbedtls/issues/4926

**Review by mpg — APPROVED** [review]
My misunderstanding about the intended scope of the task has been clarified. All looking good to me.

**Pull request merged**